numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -353,6 +353,21 @@ gentype_repr(PyObject *self)
     return ret;
 }
 
+static PyObject *
+genint_type_str(PyObject *self)
+{
+    PyObject  *item, *item_str;
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
+    }
+
+    item_str = PyObject_Str(item);
```

Comment:
Oops, I forgot this was for all types, not just int types types. Guess it doesn't do any harm though (and on python 2, makes a distinction between longs and ints)